lib/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -387,4 +396,152 @@ public boolean isVcSuccessful(List<VcStatusDto> currentVcStatuses, String criIss
+                                    .path(BIRTH_DATE_PROPERTY_NAME);
+
+                    if (birthDateNode.isMissingNode()) {
+                        LOGGER.error("BirthDate property is missing from passport VC");
+                        throw new HttpResponseExceptionWithErrorBody(
+                                500, ErrorResponse.FAILED_TO_GENERATE_IDENTIY_CLAIM);
+                    }
+
+                    List<Name> names =
+                            objectMapper.treeToValue(
+                                    nameNode,
+                                    objectMapper
+                                            .getTypeFactory()
+                                            .constructCollectionType(List.class, Name.class));
+                    List<BirthDate> birthDates =
+                            objectMapper.treeToValue(
+                                    birthDateNode,
+                                    objectMapper
+                                            .getTypeFactory()
+                                            .constructCollectionType(List.class, BirthDate.class));
+
+                    identityClaims.add(new IdentityClaim(names, birthDates));
+                } catch (ParseException | JsonProcessingException e) {
+                    LOGGER.error("Failed to parse VC JWT because: {}", e.getMessage());
+                    throw new HttpResponseExceptionWithErrorBody(
+                            500, ErrorResponse.FAILED_TO_GENERATE_IDENTIY_CLAIM);
+                }
+            }
+        }
+        return identityClaims;
+    }
+
+    public boolean checkBirthDateCorrelationInCredentials(
+            String userId, List<VcStatusDto> currentVcStatuses)
+            throws HttpResponseExceptionWithErrorBody {
+        if (currentVcStatuses != null) {
+            List<VcStoreItem> vcStoreItems = getVcStoreItems(userId);
+            List<IdentityClaim> identityClaims = getIdentityClaims(vcStoreItems, currentVcStatuses);
+            return identityClaims.stream()
+                            .map(IdentityClaim::getBirthDate)
+                            .flatMap(List::stream)
+                            .map(BirthDate::getValue)
+                            .distinct()
+                            .count()
+                    == 1;
+        }
+        return true;
+    }
+
+    public boolean checkNameAndFamilyNameCorrelationInCredentials(
+            String userId, List<VcStatusDto> currentVcStatuses)
+            throws HttpResponseExceptionWithErrorBody {
+        if (currentVcStatuses != null) {
+            List<VcStoreItem> vcStoreItems = getVcStoreItems(userId);
+
+            List<IdentityClaim> identityClaims = getIdentityClaims(vcStoreItems, currentVcStatuses);
+
+            return checkNamesForCorrelation(getFullNamesFromCredentials(identityClaims));
+        }
+        return true;
+    }
+
+    public boolean checkNamesForCorrelation(List<String> userFullNames) {
```

Comment:
looks like this method can be private.

Reply:
While it's true that the this method could potentially be made private, there are still some scenarios where it would be useful to test it in isolation as part of an integration test for this task requirements. (To compare each value, ignore case, ignore accents/diacritics, ignore spaces, ignore hyphens, ignore apostrophes.") It would be valuable to test it to ensure that those components are functioning correctly. Therefore, it can be beneficial to keep this method as a public method and include integration tests that cover different test scenarios  to validate its behaviour.